### PR TITLE
Maxpool padding

### LINKED ISSUE #31: Accuracy issue with MaxPool padding
```Python 
    def test_padding(self):
        x = np.random.rand(1, 3, 112, 112).astype(np.float32)
        model = torch.nn.Sequential(torch.nn.Conv2d(3, 64, (3, 3), 1, 1), torch.nn.MaxPool2d(3, 2, 1))
        convert_and_compare_output(model, x)
```

I isolated accuracy issues in a model to MaxPool being used with padding.
This example above is a minimum repro to fail a tolerance of 1e-3.

The problem gets worse with larger models that contain multiple such methods. My large models with over 100 layers all get complete accuracy failure due to this padding issue.

> Arrays are not almost equal to 3 decimals
> 
> Mismatched elements: 2302 / 193600 (1.19%)
> Max absolute difference: 0.82550406
> Max relative difference: 0.2062616
>  x: array([[[[ 0.301,  0.54 ,  0.498, ...,  0.476,  0.463,  0.485],
>          [ 0.822,  0.822,  0.512, ...,  0.476,  0.634,  0.634],
>          [ 0.725,  0.524,  0.512, ...,  0.469,  0.495,  0.495],...
>  y: array([[[[ 0.301,  0.54 ,  0.498, ...,  0.476,  0.463,  0.485],
>          [ 0.822,  0.822,  0.512, ...,  0.476,  0.634,  0.634],
>          [ 0.725,  0.524,  0.512, ...,  0.469,  0.495,  0.495],...
> 
> ----------------------------------------------------------------------
> Ran 1 test in 2.327s
> 
> FAILED (failures=1)

After testing this somewhat, I discovered that tf.pad works instead, with -127.5.
```Python
                paddings = ((0,0), (pads[0], pads[2]), (pads[1], pads[3]), (0,0))
                x = tf.pad(x, paddings, constant_values=-127.5)
```

Don't ask me why. The docs say it zero-pads!

I feel like it might be avoiding some internal optimisation. Perhaps in the ONNX conversion?

Before change:
Max error on my Keras model (image of George Bush): 1.9706213
After change:
Max error on my Keras model (image of George Bush): 8.895993e-06

Basically, anyone using this MaxPool op with padding is likely to have an incorrect keras model output

### PR DESCRIPTION
**Pull request by xsacha**
Fix accuracy issues

Fixes #31

### CODE REVIEW

**Comment by xsacha** [comment]
Unfortunately this does not convert to the EdgeTPU. There must be a better way.
I manually hacked in a ZeroPad2d and implemented the op_pad functionality using the Keras Zero Pad (as it was in MaxPool) and managed to get an accurate model.

**Comment by xsacha** [comment]
Due to #34 being merged, this PR is now valid.

The issue still occurs and this PR fixes it but it doesn't look good. If someone can explain why it would be nice.

**Comment by 95hali74** [comment]
What exactly doesn't look good? The extra nodes in the onnx model?

**Comment by xsacha** [comment]
The number -127.5 doesn't look good. I can't explain why it works.

**Comment by xsacha** [comment]
Any update on this?
I'm using this magic number (-127.5) in production and am happy with the end result.

**Comment by 95hali74** [comment]
I think this will work for now. But I think I found the reason why -127.5 seems to work better than 0. The inaccuracy occurs when the padding value is higher than the other values in the input tensor for a given kernel window. The maxpool op will then select the padding value rather than the other values in the input tensor.

**Comment by xsacha** [comment]
Thanks. That makes sense.
